src/server/game/Spells/SpellInfo.cpp

```diff
@@ -1346,7 +1346,7 @@ bool SpellInfo::CanPierceImmuneAura(SpellInfo const* auraSpellInfo) const
         // ...but not these (Divine shield, Ice block, Cyclone and Banish for example)
         if (auraSpellInfo->Mechanic != MECHANIC_IMMUNE_SHIELD &&
                auraSpellInfo->Mechanic != MECHANIC_INVULNERABILITY &&
-               (auraSpellInfo->Mechanic != MECHANIC_BANISH || (IsRankOf(auraSpellInfo) && auraSpellInfo->Dispel != DISPEL_NONE))) // Banish shouldn't be immune to itself, but Cyclone should
+               auraSpellInfo->Mechanic != MECHANIC_BANISH)
```

Comment:
Wihout this change, banish pierce throw banish, so script won't work

Reply:
is this needed in master too ?

Reply:
Don't test in master, but I think - yes

Reply:
could you check that there is no regression about https://github.com/TrinityCore/TrinityCore/issues/18479 and also that Cyclone still works the way it's supposed to ?

Reply:
![image](https://user-images.githubusercontent.com/11771800/62707790-10bcfd00-ba03-11e9-9fe1-617cd167eb21.png)
Works fine